### PR TITLE
Add new class scope and % pathresolver

### PR DESCRIPTION
**Pull request by tomlm**
Class scope is ActiveDialog properties.
For DCR  #2423

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Declarative/ExpressionProperty.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public ExpressionProperty(T value)
         /// </summary>
         public string Expression
         {
-            get { return expression.ToString(); }
+            get { return expression?.ToString(); }
             set { expression = new ExpressionEngine().Parse(value); }
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Dialogs.Declarative/ExpressionProperty.cs]: I know this isn't in your change, but how does this work?  This produces a value of type Expression, not string.

File: libraries/Microsoft.Bot.Builder.Dialogs/Constants/ScopePath.cs [hunk 2]
```diff
@@ -6,6 +6,7 @@ public class ScopePath
         public const string CONVERSATION = "conversation";
         public const string DIALOG = "dialog";
         public const string THIS = "this";
+        public const string CLASS = "class";
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Dialogs/Constants/ScopePath.cs]: How is "class" different from "this"?

Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.Dialogs/Constants/ScopePath.cs]: Static variables of the class?

File: libraries/Microsoft.Bot.Builder.Dialogs.Declarative/ExpressionProperty.cs (hunk 1 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs.Declarative/ExpressionProperty.cs]: yep

File: libraries/Microsoft.Bot.Builder.Dialogs/Constants/ScopePath.cs (hunk 2 above)
Comment (tomlm) [reply libraries/Microsoft.Bot.Builder.Dialogs/Constants/ScopePath.cs]: yep

**Pull request merged**